packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -0,0 +1,149 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import { providerErrors, rpcErrors } from '@metamask/rpc-errors';
+import type {
+  JsonRpcRequest,
+  ScheduleBackgroundEventParams,
+  ScheduleBackgroundEventResult,
+} from '@metamask/snaps-sdk';
+import type { CronjobRpcRequest } from '@metamask/snaps-utils';
+import {
+  CronjobRpcRequestStruct,
+  type InferMatching,
+} from '@metamask/snaps-utils';
+import {
+  StructError,
+  create,
+  object,
+  refine,
+  string,
+} from '@metamask/superstruct';
+import { type PendingJsonRpcResponse } from '@metamask/utils';
+import { DateTime } from 'luxon';
+
+import { SnapEndowments } from '../endowments';
+import type { MethodHooksObject } from '../utils';
+
+const methodName = 'snap_scheduleBackgroundEvent';
+
+const hookNames: MethodHooksObject<ScheduleBackgroundEventMethodHooks> = {
+  scheduleBackgroundEvent: true,
+  hasPermission: true,
+};
+
+type ScheduleBackgroundEventHookParams = {
+  date: string;
+  request: CronjobRpcRequest;
+};
+
+export type ScheduleBackgroundEventMethodHooks = {
+  scheduleBackgroundEvent: (
+    snapEvent: ScheduleBackgroundEventHookParams,
+  ) => string;
+
+  hasPermission: (permissionName: string) => boolean;
+};
+
+export const scheduleBackgroundEventHandler: PermittedHandlerExport<
+  ScheduleBackgroundEventMethodHooks,
+  ScheduleBackgroundEventParameters,
+  ScheduleBackgroundEventResult
+> = {
+  methodNames: [methodName],
+  implementation: getScheduleBackgroundEventImplementation,
+  hookNames,
+};
+
+const offsetRegex = /Z|([+-]\d{2}:?\d{2})$/u;
+const ScheduleBackgroundEventsParametersStruct = object({
+  date: refine(string(), 'date', (val) => {
+    const date = DateTime.fromISO(val);
+    if (date.isValid) {
+      // luxon doesn't have a reliable way to check if timezone info was not provided
+      if (!offsetRegex.test(val)) {
```

Comment:
Given this, what is the expected way for Snaps to use the API if they don't wanna care about timezones?

I asked a question elsewhere whether we should allow omitting the timezone which you disagreed with.

Reply:
They're going to have to care about timezones to schedule an event, validation will not pass otherwise.

Reply:
I guess I should have phrased it differently. What is the most low effort way to schedule a timer 30 secs from now?

I'm just trying to determine if this API will be useful for the use-cases we have in mind!

Reply:
Not the lowest of efforts but the process would be get the current date, add 30 seconds, convert to UTC, convert that to an ISO 8601 string and schedule with that date.

I don't mind writing a util that devs can use. Maybe something like `function getISODateIn(amount, unit)`